src/TMainConsole.cpp

```diff
@@ -155,12 +155,15 @@ std::pair<bool, QString> TMainConsole::setCmdLineStyleSheet(const QString& name,
 
 void TMainConsole::toggleLogging(bool isMessageEnabled)
 {
-    QFile file(mudlet::getMudletPath(mudlet::profileDataItemPath, mpHost->getName(), qsl("autolog")));
+    const auto loggingPath = mudlet::getMudletPath(mudlet::profileDataItemPath, mpHost->getName(), qsl("autolog"));
+    QSaveFile file(loggingPath);
```

Comment:
Can creating the file fail?

Reply:
Well no more than it can with the conventional `QFile` - the important difference about the `QSaveFile` class is that it either completes and overwrites the entire contents of a previous save file instance or doesn't and leaves the prior instance untouched - but there are NO contents to be left in an inconsistent state by a partial/incomplete write in this situation so there is no benefit to using it...

Reply:
Sure, reverted.

Reply:
Actually no, I reverted the revert. We still can have a failure to open the file for writing for whatever reason.

Reply:
Which the existing `QFile` class is perfectly capable of handling...

Reply:
Sure - but the current code didn't - and with qsavefile it does. It's a bit overkill, I agree.